src/analysis/retail/evoker/preservation/modules/core/MasteryEffectiveness.tsx

```diff
@@ -266,6 +235,73 @@ class MasteryEffectiveness extends Analyzer {
     this.totalMasteryHealingDone += absorbFromMasteryBonusUsed;
   }
 
+  __masteryBoostCheck(event: ApplyBuffEvent | RefreshBuffEvent, existingAbsorb: number) {
+    const mastery = this.statTracker.currentMasteryPercentage;
+    const baseShieldValue = this.__calculateShield(event, existingAbsorb);
+    const lowerShieldBound = baseShieldValue * (1 + (mastery - 0.1));
+    const upperShieldBound = baseShieldValue * (1 + (mastery + 0.1));
+    if (event.absorb) {
+      if (lowerShieldBound <= event.absorb && event.absorb <= upperShieldBound) {
+        return true;
+      }
+    }
+    return false;
+  }
+  //TODO -- THESE CALCULATIONS ARE USING A +/- 10% THRESHOLD UNTIL ALL THE NEW RAID BUFFS ARE ADDED AND INCORPORATED INTO STATTRACKER
+  __calculateShield(event: ApplyBuffEvent | RefreshBuffEvent, existingAbsorb: number) {
+    let baseShielding = 0;
+    const vers = this.statTracker.currentVersatilityPercentage + 0.03; // TODO: REMOVE WHEN MARK OF THE WILD IS ADDED TO STATTRACKER
+    if (this.__isEventAbilityAnomaly(event.ability.guid)) {
+      const intellect = this.statTracker.currentIntellectRating;
+      baseShielding = intellect * TEMPORAL_ANOMALY_SHIELD_COEFF * (1 + vers) + existingAbsorb;
+    }
+
+    if (this.__isEventAbilityTwinGuardian(event.ability.guid)) {
+      const initialShieldAmount = this.prevokerMaxHealth * TWIN_GUARDIAN_SHIELD_COEFF;
+      const versAddedShield = this.prevokerMaxHealth * TWIN_GUARDIAN_SHIELD_COEFF * vers;
+      baseShielding = initialShieldAmount + versAddedShield + existingAbsorb;
+    }
```

Comment:
these shields stack with themselves? trying to understand what you're doing with the existing absorb value here and that's what I think is going on.

Reply:
Yeah the two don't stack with each other, but Temporal Anomaly stacks on itself which is fairly common as it pulses in a small radius every 2s. The refresh value needs to be evaluated with any existing shields added back in or its way off